src/libaktualizr/logging/boost_logging.cc

```diff
@@ -25,8 +25,26 @@ int loggerGetSeverity() { return static_cast<int>(gLoggingThreshold); }
 
 void LoggerConfig::updateFromPropertyTree(const boost::property_tree::ptree& pt) {
   CopyFromConfig(loglevel, "loglevel", boost::log::trivial::trace, pt);
+  // If not already set from the commandline, set the loglevel now so that it
+  // affects the rest of the config processing.
+  if (!initialized) {
+    setLogLevel();
```

Comment:
This bugs me a little bit that `setLogLevel` is a method of LoggerConfig and that the `updateFromPropertyTree` does more that it says here (and departs from the other `xxConfig` objects). What about moving `initialized` and `setLogLevel` out of the class?

Maybe, make it possible to call `logger_init()` several times, with an optional `LoggerConfig` argument and use it inside `Config::updateFromPropertyTree`

Reply:
I don't think having extra functionality in `LoggerConfig` is inherently bad, but I do see that setting the loglevel from inside `updateFromPropertyTree()` is a bit odd. What's the benefit of calling `logger_init()` several times, though? We already have `logger_set_threshold()` that can be called multiple times.

I suppose I could make setLogLevel() a non-member function (maybe even just overload `logger_set_threshold()` with an integer input). The `initialized` variable annoys me but I can't find a good way to get around it.

Reply:
`setLogLevel` should not be method for the config: it primary applies to the logging library itself and only incidently reads the config. The usual pattern is that the config objects are dumb and don't depend on the module they apply to. On the contrary, the module depends on the config.

`initialized` can be a static variable outside of the class. It's slightly ugly but logging facilities are the kind of canonical examples of acceptable uses of a singleton.

I thought about reusing `logger_init()` because it is a good name for a function that configures the library entirely from a configuration object.